spec/models/invoice_spec.rb

```diff
@@ -1765,4 +1765,45 @@
       end
     end
   end
+
+  describe "should_apply_provider_tax?" do
+    let(:invoice) { create(:invoice, :subscription, subscriptions: [subscription]) }
+    let(:plan) { create(:plan) }
+    let(:subscription) { create(:subscription, plan:) }
+
+    let(:charge) { create(:standard_charge, plan:) }
+    let(:charge_fee) { create(:charge_fee, charge:, invoice:, amount: 100) }
+
+    before { charge_fee }
+
+    context "when there are fees with non zero amount" do
+      it { expect(invoice).to be_should_apply_provider_tax }
```

Comment:
Another a matter of taste thing: keep contexts symmetry.

```suggestion
    before { charge_fee }

    context "when there are fees with non zero amount" do
      let(:charge_fee) { create(:charge_fee, charge:, invoice:, amount: 100) }

      it { expect(invoice).to be_should_apply_provider_tax }
```